x-pack/plugins/alerts/server/alerts_client.ts

```diff
@@ -421,6 +423,14 @@ export class AlertsClient {
   }
 
   public async update({ id, data }: UpdateOptions): Promise<PartialAlert> {
+    return await retryIfConflicts(
+      this.logger,
+      `alertsClient.update('${id}')`,
+      async () => await this.updateWithOCC({ id, data })
+    );
+  }
+
+  private async updateWithOCC({ id, data }: UpdateOptions): Promise<PartialAlert> {
```

Comment:
Seeing as AlertsClient is already a huge file, is there a way we can add the retry into this without adding an extra method for each operation? 🤔 
Perhaps by using some kind of _decorator_ that wraps these methods or something along those lines?
This will also become quite a bit more complicated when we introduce optional `version` to these methods, if we can find a way to reduce this into a single extension, it will help with maintenance down the line.

Reply:
Do we support decorators?  I'm not sure inlining these is going to save all that space, is it?  My thinking was that once we add versions to these methods, that basically just means to set retries to 0, so will likely be a straight-forward change to each of the sites.

The reason I separated these out was to try to keep the actual methods a little cleaner - wrapping like this obviously doesn't scale real well, beyond the first time to you do it (now!).  I can go either way on inlining, but I think the current shape is a little more readable than inlined.

Reply:
After conversation with Gidi, we decided to think about some alternatives to this approach, but couldn't come up with anything that would require even more changes to the alerts client.